### PR TITLE
fix bugs related to no_std

### PR DESCRIPTION
**Pull request by anonrig**

### CODE REVIEW

**Review by ObsidianMinor — APPROVED** [review]
Probably should include cargo-hack in CI it seems

File: recapn/src/io.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub enum ReadError {
     Incomplete,
 }
 
+#[cfg(feature = "std")]
```
Comment (ObsidianMinor) [inline recapn/src/io.rs:51]: This should work under `no_std`...

Comment (anonrig) [reply recapn/src/io.rs]: Yeah, I removed it for now. We need cargo-hack to make sure this thing actually works.

**Comment by anonrig** [comment]
@ObsidianMinor would you mind reviewing and landing this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/checkout@v4
       - name: Install latest stable Rust
         uses: dtolnay/rust-toolchain@stable
-      - name: Test no default features
-        run: cargo test --all --no-default-features
-      - name: Test all features
-        run: cargo test --all --all-features
+      - name: Install cargo-hack
```
Comment (mikea) [inline .github/workflows/ci.yml:14]: awesome!

File: recapn/src/alloc.rs [hunk 3]
```diff
@@ -442,7 +442,8 @@ impl<const N: usize> Space<N> {
 ///
 /// # Example
 ///
-/// ```
+#[cfg_attr(feature = "alloc", doc = " ```")]
+#[cfg_attr(not(feature = "alloc"), doc = " ```ignore")]
```
Comment (ObsidianMinor) [inline recapn/src/alloc.rs:446]: I should really just fix this test so it doesn't require Message... And this looks really bad so for now I'm going to suggest changing it to
```rust
/// ```
/// # #[cfg(feature = "alloc")] {
/// use recapn::alloc;
/// use recapn::message::Message;
///
/// let mut space = alloc::space::<16>();
/// let message = Message::with_scratch(&mut space);
/// # drop(message); }
/// ```
```

Comment (anonrig) [reply recapn/src/alloc.rs]: I've pushed a new commit for this. Thanks!

**Pull request merged**